tiledb/consolidation_plan.py

```diff
@@ -0,0 +1,59 @@
+import tiledb.cc as lt
+
+from .ctx import Ctx, CtxMixin, default_ctx
+from .libtiledb import Array
+
+
+class ConsolidationPlan(CtxMixin, lt.ConsolidationPlan):
+    """Represents TileDB ConsolidationPlan class"""
+
+    def __init__(self, ctx: Ctx, array: lt.Array, fragment_size: int):
+        """Class representing the consolidation plan for an array. The plan divides the fragments of the array into different nodes that need to be consolidated together.
+
+        :param ctx: A TileDB Context
+        :param array: The array
+        :param fragment_size: The desired fragment size
+        :raises TypeError: All dimensions must have the same dtype
```

Comment:
this needs an update?

Reply:
Why? I have already added a more detailed description as Vivian proposed. Arguments seem fine.

Reply:
"TypeError: All dimensions must have the same dtype" doesn't seem to me like an error Consolidation Plan would throw.